### ISSUE TITLE
[Rework] Replace Occupations and antag systems with "Character Attributes"

### ISSUE THREAD

**Issue by MaxIsJoe** [unitystation/unitystation#9239]
While planning out my work on Fallout 13's experiment in unitystation I realized that our current way to assign occupations to people is kind of flawed and limiting in some way. As they're designed currently, they are not fork friendly at all and always assume that all forks will follow the /tg/ design when it comes to jobs and antagonists. I've come up with a solution for this and a way to add features that are missing from unitystation more easily,

# The solution:

Characters have a list of "Attributes" that let them spawn with quirks that define who they are and what they can (or can't) do. Attributes are generic and serve as a way to shape a character's identity, both gameplay and role-play wise.

In short, attributes can be used as a way to label a character as something or give them special functionality.

Due to their generic nature, this allows forks to design their server's characters in the way that seems fit for them without having to re-program a lot of stuff from scratch.

This also helps us clean up our code a little by unifying our systems a tad bit and make them less scattered around.

### Example for use cases:
A) Player characters can have the same jobs, but each character belongs to a different faction. Both their jobs and factions are assigned as an Attribute.
B) Admins want to create a nukie wizard, they give a player both the "wizard" and "nuke ops" Attributes. (Mix and mash different roles together)
C) Server wants to remove occupations/jobs/antags all together and want characters to spawn with “backgrounds”, like an RPG. Characters are given attributes that setups up their character based on the background the player chooses.
D) Zombies that bite other players can give them the “infected” attribute.
E) Allows us to create misc roles for gameplay/roleplay reasons. (I.e : Chaplain who has to feed holy water to a clown with the 'possessed/unblessed' attribute)
F) Adding character traits/quirks similar to the ones on /tg/ (Heavy smoker, paraplegic, Pushover, etc)

_Note: this will also solve one of our issues with #8036 as this solution easily lets us control all characters' role status_
_This also allows us to add this -> https://tgstation13.org/wiki/Quirks_

# Pseudo coding:

```cs

class CharacterAttribute : ScriptableObject{

//General data that will be on every Attribute
string AttributeTitle = "";
string AttributeDesc = "";
//Attributes that can't go hand in hand with this one
List<CharacterAttribute> Blacklist = new List<CharacterAttribute>();

//what to do with the Attribute gets added to the character
virtual void OnAdd() { }
//what to do after the Attribute gets removed from the character
virtual void OnRemove() { }
//what to do when the round ends
virtual void OnRoundEnd() { }
//what to do when the player is given this Attribute when the round starts
virtual void OnRoundStart() { }
//what to do when is given this Attribute when he joins mid-round
virtual void OnRoundJoin { }
```

```cs
//classic /tg/ and goon + city13 occupations
//basically copy paste all the variables and functions we have in our occupation code here and it should work the same
//do the same for antags
class Occupation : CharacterAttribute { }
class SpecialClown : Occupation { }

//classic /tg/ and goon antags
class Antag : CharacterAttribute { }
class Wizard : Antag { }
class Traitor: Antag { }

//Fallout 13
class Faction : CharacterAttribute { }
class BrotherhoodOfSteel : Faction { }
class Raiders : Faction { }
class NCR : Faction { }
class Profession : CharacterAttribute { }
class Wanderer : Profession { }
class Bartender : Profession { }
class Mercanary : Profession { }

//SCP13
class FoundationStaff : CharacterAttribute { }
class SiteStaff : FoundationStaff { }
class Researcher: SiteStaff { }
class Administration : FoundationStaff { }
class O5CouncilMember : Administration { }
class SiteDirector: Administration { }

//random fun ideas
class SpeaksWithGhosts : CharacterAttribute { }
class ExConvict : CharacterAttribute { }
class Merhcant : CharacterAttribute { }
class AI_Control : CharacterAttribute { }
class SicknessImmune : CharacterAttribute { }
class Blessed : CharacterAttribute { }
class BusinessMan : CharacterAttribute { }
class ExtraStrong : CharacterAttribute { }
class SpawnWithItem : CharacterAttribute { }
class SpawnItemOnDeath : CharacterAttribute { }
```

```cs
//In PlayerScript.cs
List<CharacterAttribute> CharacterAttributes = new List<CharacterAttribute>();

void AddAttribute(CharacterAttribute newRole) { }
void RemoveAttribute(CharacterAttribute targetRole) { }
```

# Notes:

- "Wizard Nukies" are just nukies who spawned and were given the "wizard" attribute and ended up receiving a spell book and a scroll.
- This doesn't add anything "new" for unitystation. It just allows developers to have an easier time adding features and for forks to change up their roles for their characters more easily.

**Comment by Bod9001** [comment]
imo I think stuff like paraplegic should be a genetic disorder?

**Comment by corp-0** [comment]
> imo I think stuff like paraplegic should be a genetic disorder?

All the Attribute would do is setting your character up during spawn and give you the genetic disorder, so it is fine

**Comment by MaxIsJoe** [comment]
This is slightly outdated now so I'm closing this in favor for a new issue that I'm going to make in a couple of days after I've gathered some more info on some stuff + bug reports after #9281 gets merged.
